src/lib/utils/npm-user-agent-parser.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Parses a user agent string for NPM metadata.
+ * @param {string} userAgent 
+ * @return {{npmVersion: string, nodeVersion: string, platform: string, architecture: string}}
+ */
+const npmUserAgentParser = (userAgent) => {
```

Comment:
Done. We need to create the scaffolding for it. I'll let you know when is ready.